lib/tom_queue/deferred_work_set.rb

```diff
@@ -8,6 +8,8 @@ module TomQueue
   #
   class DeferredWorkSet
```

Comment:
I wouldn't want to - lets just make the smallest change to break out the deferred handler.